web-local/src/lib/svelte-query/invalidation.ts

```diff
@@ -100,20 +100,30 @@ export const invalidateMetricsViewData = (
   queryClient: QueryClient,
   metricsViewName: string
 ) => {
+  // remove inactive queries, this is needed since these would be re-fetched with incorrect filter
+  queryClient.removeQueries({
+    predicate: (query) =>
+      invalidationForMetricsViewData(query, metricsViewName),
+    active: false,
+  });
   return queryClient.refetchQueries({
```

Comment:
Why not instead use `queryClient.invalidateQueries()`? By default, it refetches only the active queries.